### DISCUSSION TITLE
CI CD deployment of a custom module

### DISCUSSION THREAD

**Discussion by ghost** [OrchardCMS/OrchardCore#14067] [Q&A]
Hi, I have a custom Orchard Module that is deployed as part of a Orchard Core application in Azure. I have an App_Data folder that has site configuration details in it, but nothing around Modules. How would I go about redeploying an updated version of my module?

**Comment by ns8482e** [comment]
In OC Web project - module can be added as either Project Reference or as Nuget Package. 

If your module project is directly referenced in Web project then CI/CD of your web project will build/deploy your module.

If you have module referenced as nuget package then - you could create CI/CD pipeline for you module that publish your packages to nuget artifacts repository. And update your web project with latest version of your module from your nuget artifacts repository and build/deploy your web project via CI / CD pipeline.
